timemachine/fe/rbfe.py

```diff
@@ -702,18 +751,63 @@ def estimate_relative_free_energy_bisection_hrex_impl(
     lambda_max: float,
     md_params: MDParams,
     n_windows: int,
-    make_optimized_initial_state_fn: Callable[[float], InitialState],
+    make_initial_state_fn: Callable[[float, bool], InitialState],
```

Comment:
nit/non-blocking: idea for an alternative in case adding an argument to the callback seems unwieldy: could this accept a separate `optimize_initial_state: Callable[[InitialState], InitialState]` argument, which is only called during bisection but not refinement?

Reply:
Hmmm that seems unwieldy to me as well especially since one function would be optional depending on the setup of MDParams.

What about having a `make_initial_state_fn` and a `optimize_initial_state_fn`? That way we don't have two functions to produce the same type of object with the nuance of being optimized? Also the `optimize_initial_state_fn` could be the identity function if we wanted.

Reply:
> What about having a make_initial_state_fn and a optimize_initial_state_fn? That way we don't have two functions to produce the same type of object with the nuance of being optimized? Also the optimize_initial_state_fn could be the identity function if we wanted.

I think this is what I intended to suggest - i.e. accept two function arguments, one with type `float -> InitialState` and another with type `InitialState -> InitialState` (that could be the identity).

Reply:
Woops, that is what you suggested and I just misread it.